### PR TITLE
[jenkins] Support authenticated access using user/API token

### PR DESCRIPTION
**Pull request by valeriocos**
This PR allows to fetch builds and jobs available via a Jenkins server, which requires authenticated access.
Tests have been added accordingly. The backend version has been set to 0.12.0.

### CODE REVIEW

File: tests/test_jenkins.py [hunk 1]
```diff
@@ -311,6 +340,62 @@ def test_fetch_depth_2(self):
         self.assertRegex(req.path, '/ci/job')
         self.assertDictEqual(req.querystring, expected)
 
+    @httpretty.activate
+    def test_fetch_auth_api_token(self):
```
Comment (sduenas) [inline tests/test_jenkins.py]: I don't see anything checking that the client is really sending the authentication as is expected.

File: tests/test_jenkins.py [hunk 2]
```diff
@@ -507,6 +570,21 @@ def test_get_jobs(self):
 
         self.assertEqual(response, body)
 
+    @httpretty.activate
+    def test_get_jobs_auth_api_token(self):
```
Comment (sduenas) [inline tests/test_jenkins.py]: I think this should check that the token is sent.

**Comment by valeriocos** [comment]
> Besides my comments I miss tests for client.py and the new parameter in fetch.

There are no tests for the params in fetch (beyond GET and POST), so this is the reason why they weren't added. I'll add the missing tests in a different commit.

File: tests/test_jenkins.py (hunk 2 above)
Comment (valeriocos) [reply tests/test_jenkins.py]: The token is sent inside a basic authentication, maybe an assertion can be added to check whether in the list of headers there is `Authorization` one (which is must be `Basic`)

**Comment by sduenas** [comment]
> > Besides my comments I miss tests for client.py and the new parameter in fetch.
> 
> There are no tests for the params in fetch (beyond GET and POST), so this is the reason why they weren't added. I'll add the missing tests in a different commit.

It can be integrated in one of the test we have now. Checking the header as you said before should be enough.

**Comment by valeriocos** [comment]
check it again @sduenas

**Comment by dai-mk** [comment]
@valeriocos @sduenas 
How do you use authenticated access? I couldn't find it in the documentation. https://github.com/chaoss/grimoirelab-sirmordred#jenkins-

**Pull request merged**